pkg/system/phase4_configuring.go

```diff
@@ -597,6 +608,82 @@ func (r *Reconciler) prepareAzureBackingStore() error {
 	return nil
 }
 
+func (r *Reconciler) prepareGCPBackingStore() error {
+	cloudCredsSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      r.GCPCloudCreds.Spec.SecretRef.Name,
+			Namespace: r.GCPCloudCreds.Spec.SecretRef.Namespace,
+		},
+	}
+
+	util.KubeCheck(cloudCredsSecret)
+	if cloudCredsSecret.UID == "" {
+		// TODO: we need to figure out why secret is not created, and react accordingly
+		// e.g. maybe we are running on AWS but our CredentialsRequest is for GCP
+		r.Logger.Infof("Secret %q was not created yet by cloud-credentials operator. retry on next reconcile..", r.GCPCloudCreds.Spec.SecretRef.Name)
+		return fmt.Errorf("cloud credentials secret %q is not ready yet", r.GCPCloudCreds.Spec.SecretRef.Name)
+	}
+	r.Logger.Infof("Secret %s was created successfully by cloud-credentials operator", r.GCPCloudCreds.Spec.SecretRef.Name)
+
+	util.KubeCheck(r.GCPContainerCreds)
+	if r.GCPContainerCreds.UID == "" {
+		r.GCPContainerCreds.StringData = cloudCredsSecret.StringData
+		r.Own(r.GCPContainerCreds)
+		if err := r.Client.Create(r.Ctx, r.GCPContainerCreds); err != nil {
+			return fmt.Errorf("got error on GCPContainerCreds creation. error: %v", err)
+		}
+	}
+	bytes, err := b64.StdEncoding.DecodeString(string(cloudCredsSecret.Data["service_account.json"])) // Converting data
```

Comment:
unmarshal the cloudCredsSecret.Data["service_account.json"] rather than decode base64.. see https://github.com/openshift/cloud-credential-operator/blob/f7e71249b9d8c6d9135c035142b7057443a6676c/pkg/gcp/actuator/utils.go#L61

Reply:
Updated!.